pipeline/transformer/transformer_calm/src/main/scala/weco/pipeline/transformer/calm/transformers/CalmMergeCandidates.scala

```diff
@@ -0,0 +1,42 @@
+package weco.pipeline.transformer.calm.transformers
+
+import weco.catalogue.internal_model.identifiers.{
+  IdState,
+  IdentifierType,
+  SourceIdentifier
+}
+import weco.catalogue.internal_model.work.MergeCandidate
+import weco.catalogue.source_model.calm.CalmRecord
+import weco.pipeline.transformer.calm.models.CalmRecordOps
+
+object CalmMergeCandidates extends CalmRecordOps {
+  def apply(record: CalmRecord): List[MergeCandidate[IdState.Identifiable]] =
+    sierraMergeCandidate(record).toList ++ miroMergeCandidates(record)
+
+  private def miroMergeCandidates(record: CalmRecord) =
+    // The internal field "Wheels" is mapped to "MiroID"
+    record.getList("Wheels").map { miroIdentifier =>
+      MergeCandidate(
+        IdState.Identifiable(
+          SourceIdentifier(
+            identifierType = IdentifierType.MiroImageNumber,
+            ontologyType = "Work",
+            value = miroIdentifier
+          )
+        )
+      )
+    }
+
+  private def sierraMergeCandidate(record: CalmRecord) =
+    record.get("BNumber").map { bNumber =>
+      MergeCandidate(
+        IdState.Identifiable(
+          SourceIdentifier(
+            identifierType = IdentifierType.SierraSystemNumber,
+            ontologyType = "Work",
+            value = bNumber
+          )
+        )
+      )
+    }
```

Comment:
I think, both here and for the Miro ID, that we should so some validation on what's in this field, and skip values that don't look exactly like a Miro ID or a Sierra ID – possibly dropping a warning log if they're not right.

e.g. [6b9200c4-91e8-45dd-9085-1b0834ff1bdd](https://reporting.wellcomecollection.org/s/calm/app/discover#/doc/90cfbc80-d19f-11ea-8175-efb55f54ba0b/calm_catalog?id=6b9200c4-91e8-45dd-9085-1b0834ff1bdd)

```
Wheels
<a href="http://archives.wellcomelibrary.org/DServe/dserve.exe?&dsqIni=Dserve.ini&dsqApp=Archive&dsqCmd=show.tcl&dsqDb=Catalog&dsqPos=0&dsqSearch=(AltRefNo='WA/HSW/PH/A.3')" target="_blank">Click here to see the description of this item in the full archive catalogue</a>
```

which is pretty clearly not a Miro ID.

Reply:
Also, be aware that SourceIdentifier will throw an IllegalArgumentException if you pass it certain invalid values, e.g. with leading/trailing whitespace. In that case we should ignore the merge candidate and transform the Work anyhow – which you could either fix by wrapping this in a `Try` block, or by filtering out illegal values before you pass them into SourceIdentifier.

Reply:
Strangely, the example you link to is not actually how that record appears in the CALM API (the live version doesn't have a `Wheels` field, and its `Modified` date is `19/03/2021` rather than `26/03/2020`) - is there something up with the way CALM data gets into the reporting cluster?

Reply:
That seems likely – I've had at least one other case where the reporting data is wrong.

Filed a ticket for it: https://github.com/wellcomecollection/platform/issues/5240